### PR TITLE
DAGs for MPX data and Population Data. 

### PR DESCRIPTION
**Pull request by NicoRicardi**
Currently, variable folder refers to my machine. How do we put it to work for the egh server? (relates to MPX-dashboard)

### CODE REVIEW

File: docker/airflow/dags/monkeypox_dag.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+                               load_cases_kwargs=load_cases_kwargs,
+                               to_parquet_kwargs=to_parquet_kwargs),
+                       folder,
+                       read_parquet_kwargs=read_parquet_kwargs,
+                       conf_aggr=conf_aggr,
+                       sus_aggr=sus_aggr)
+
+cols_needed = [csv_specs.countrycol, csv_specs.confdatecol, csv_specs.deathdatecol, csv_specs.entrydatecol, csv_specs.statuscol]
+dag = etl_mpx('https://raw.githubusercontent.com/globaldothealth/monkeypox/main/latest_deprecated.csv',
+              folder='/home/nr/WORK/MPX-dashboard/data',
```
Comment (fccoelho) [inline docker/airflow/dags/monkeypox_dag.py:178]: use a subdir of `/tmp` here that's what other Data collection scripts do. They actually configure this sort of things [here](https://github.com/thegraphnetwork/epigraphhub_py/blob/main/epigraphhub/data/data_collection/config.py)

File: docker/airflow/dags/monkeypox_dag.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+        date = pd.to_datetime('today').strftime('%d_%m_%Y')
+        path = os.path.join(folder, f'{date}.parquet')
+        raw_data.to_parquet(path, **to_parquet_kwargs)
+        logging.info(f'raw data saved in {path}')
+        return path
+
+    @task
+    def transform_and_load(path: str, folder: str, read_parquet_kwargs: dict = {},
+                           conf_aggr: dict = {}, sus_aggr: dict = {}):
+        """
```
Comment (fccoelho) [inline docker/airflow/dags/monkeypox_dag.py:108]: Before the list of parameters add a short but informative description of the purpose of this function

File: docker/airflow/dags/population_dag.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+    Returns
+    -------
+    None
+    """
+
+    @task
+    def extract_and_load(all_countries: list, folder: str = '',
+                         type_: str = 'country', output: str = 'population'):
+        """
```
Comment (fccoelho) [inline docker/airflow/dags/population_dag.py:48]: Again missing description on the Docstring

File: docker/airflow/dags/population_dag.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+            logging.info('Population loaded!')
+        except Exception as e:
+            tback = traceback.format_exc()
+            logging.critical(f'loading raw data failed. Error message and traceback below.\n {e} \n {tback}')
+        output = os.path.join(folder, output if output.endswith('.csv') else f'{output}.csv')
+        population.to_csv(os.path.join(folder,f'{output}.csv')) 
+    
+    extract_and_load(all_countries, folder=folder, type_=type_, output=output)
+    
+dag = population_dag(utils.all_countries, folder='/home/nr/WORK/MPX-dashboard/data')
```
Comment (fccoelho) [inline docker/airflow/dags/population_dag.py:78]: As before, use `/tmp` as path

**Review by fccoelho — CHANGES_REQUESTED** [review]

File: docker/airflow/dags/monkeypox_dag.py (hunk 1 above)
Comment (NicoRicardi) [reply docker/airflow/dags/monkeypox_dag.py]: I really do not think that it should be a temporary folder, we ought to keep a copy of this raw data. I doubt the size of the files is a problem, but if it were, we can store a copy with a lower frequency and/or check if and what was modified. 
I have changed the env.tpl and and the main.yaml to have an environment variable, and the dag to use the environment variable. The name or location can of course be discussed and changed.

Ivan told me he will open a PR to do a minIO.

File: docker/airflow/dags/monkeypox_dag.py (hunk 2 above)
Comment (NicoRicardi) [reply docker/airflow/dags/monkeypox_dag.py:121]: Done!

File: docker/airflow/dags/population_dag.py (hunk 3 above)
Comment (NicoRicardi) [reply docker/airflow/dags/population_dag.py:54]: Done!

File: docker/airflow/dags/population_dag.py (hunk 4 above)
Comment (NicoRicardi) [reply docker/airflow/dags/population_dag.py]: This file is the final result of the DAG (in the MPX, the first file is a partial result), so I am not sure of why we would want to put it in a temporary folder. We need to store it.

The frequency can be changed from monthly to something between monthly and yearly (e.g. 2-4 months), if we want to minimise operations.

Comment (fccoelho) [reply docker/airflow/dags/population_dag.py]: for permanent storage we need to ask @xmnlab  to define a path on  the server that can be user for this purpose until we have the  `minio` service up and running. So for now I suggest skipping the storage of the CSV and just inserting the data in the database.

File: docker/airflow/dags/monkeypox_dag.py [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+        Note
```
Comment (fccoelho) [inline docker/airflow/dags/monkeypox_dag.py:122]: you can remove this "Note" header.

File: docker/airflow/dags/population_dag.py (hunk 4 above)
Comment (xmnlab) [reply docker/airflow/dags/population_dag.py]: let's start already with minio. 
I can open a PR for that today. it should be easy.

**Comment by xmnlab** [comment]
@NicoRicardi do you mind rebasing your code on top of the main branch pls? thanks!

File: docker/airflow/dags/monkeypox_dag.py (hunk 1 above)
Comment (fccoelho) [reply docker/airflow/dags/monkeypox_dag.py]: MinIO is coming online this weekend @NicoRicardi. I hope we can finish this last issue and merge this PR next week.

**Comment by luabida** [comment]
@NicoRicardi @fccoelho 
I couldn't find a reference to install the package `mpx_dashboard` in the dependencies, so I've tried to install manually using the command `pip install git+https://github.com/thegraphnetwork/MPX-dashboard` inside the container. But when I tried to import the module `mpx_dashboard`, there were no other modules to use. The return of `dir(mpx_dashboard)` is:
```
>>> import mpx_dashboard
>>> dir(mpx_dashboard)
['__builtins__', '__cached__', '__doc__', '__file__', '__loader__', '__name__', '__package__', '__path__', '__spec__']
```

Then I tried to clone the repository into the container and installing using `pip install -e`, which also didn't work, the directory of the installed package was empty.

How should the `mpx_dashboard` package be installed?

**Comment by xmnlab** [comment]
@fccoelho any plans for this PR?

**Comment by fccoelho** [comment]
closing as no longer relevant

**Pull request closed without merging**